feat: support new tags attribute in cluster, advanced_cluster, and serverless_instance (#1461)
mongodb/terraform-provider-mongodbatlas: website/docs/r/cluster.html.markdown

  * Cannot be used with clusters with local NVMe SSDs
  * Cannot be used with Azure clusters
* `encryption_at_rest_provider` - (Optional) Possible values are AWS, GCP, AZURE or NONE.  Only needed if you desire to manage the keys, see [Encryption at Rest using Customer Key Management](https://docs.atlas.mongodb.com/security-aws-kms/) for complete documentation.  You must configure encryption at rest for the Atlas project before enabling it on any cluster in the project. For complete documentation on configuring Encryption at Rest, see Encryption at Rest using Customer Key Management. Requires M10 or greater. and for legacy backups, backup_enabled, to be false or omitted. **Note: Atlas encrypts all cluster storage and snapshot volumes, securing all cluster data on disk: a concept known as encryption at rest, by default**.   

> * `tags` - (Optional) Configuration for the collection of key-value pairs that tag and categorize the cluster. See [below](#tags).

Reviewer: ```suggestion
* `tags` - (Optional) List that contains key-value pairs between 1 to 255 characters in length for tagging and categorizing the cluster. See [tags](#tags).
```

> * `labels` - (Optional) Configuration for the collection of key-value pairs that tag and categorize the cluster. See [below](#labels). **DEPRECATED** Use `tags` instead.

Reviewer: ```suggestion
* `labels` - (Optional) Configuration for the collection of key-value pairs that tag and categorize the cluster. See [labels](#labels). **DEPRECATED** Use `tags` instead.
```

* `sample_refresh_interval_bi_connector` - (Optional) Interval in seconds at which the mongosqld process re-samples data to create its relational schema. The default value is 300. The specified value must be a positive integer. Available only for Atlas deployments in which BI Connector for Atlas is enabled.
* `transaction_lifetime_limit_seconds` - (Optional) Lifetime, in seconds, of multi-document transactions. Defaults to 60 seconds.

### Tags

 ```terraform
 tags {
        key   = "Key 1"
        value = "Value 1"
  }
 tags {
        key   = "Key 2"
        value = "Value 2"
  }
```

>  Key-value pairs used for tagging and categorizing the cluster. Each key and value has a maximum length of 255 characters.

Reviewer: ```suggestion
Key-value pairs between 1 to 255 characters in length for tagging and categorizing the cluster.
```

* `key` - (Required) Constant that defines the set of the tag.
* `value` - (Required) Variable that belongs to the set of the tag.

> For additional information you can reference [Resource Tags](https://www.mongodb.com/docs/atlas/tags/).

Reviewer: ```suggestion
To learn more, see [Resource Tags](https://dochub.mongodb.org/core/add-cluster-tag-atlas).
```

### Labels

> **WARNING:** This property is deprecated and will be removed by November 2023, use `tags` attribute instead.

Reviewer: ```suggestion
**WARNING:** This property is deprecated and will be removed by November 2023, use the `tags` attribute instead.
```
